actionpack/lib/action_dispatch/testing/request_encoder.rb

```diff
@@ -51,5 +51,6 @@ def self.register_encoder(mime_name, param_encoder: nil, response_parser: nil)
     end
 
     register_encoder :json, response_parser: -> body { JSON.parse(body) }
+    register_encoder :html, response_parser: -> body { Nokogiri::HTML5.parse(body) }
```

Comment:
We need to require nokogiri in this file, because nokogiri might not be required yet. This also means that `nokogiri` will need to be added as dependency of actionpack, that it isn't right now.

Reply:
It is fine to add the dependency, it is already an indirect dependency. We just need to make it explicit.

Reply:
Does that require a change to the gemspec, or would a `require "nokogiri"` line suffice?

Reply:
Need to change the gemspec as well.

Reply:
Done!